src/node_modules.cc

```diff
@@ -326,6 +330,22 @@ const BindingData::PackageConfig* BindingData::TraverseParent(
   return nullptr;
 }
 
+#ifdef _WIN32
+std::wstring ConvertToWideString(const std::string& str) {
+  auto cp = GetACP();
```

Comment:
This section was copied from node_file.cc, but this `GetACP` is particularly important.
As shown in the description, when the Windows locale is changed and executed, if the original UTF-8 code is left here, the strings cannot be handled properly.

Reply:
I wonder if this utility function can be moved to `util-inl.h`? so that the same implementation that takes into account `GetACP` is used both here and in `node_file.cc` without duplication. I think this is also what @anonrig is requesting.

Reply:
If we do the same for node_fs, test-fs-cp.mjs will falil...

It is very strange, but I have found that if I run the failed path(utf/~~~) in addition to the test-require-~ , it can be handled correctly.
It may be that the pre or post-process (if there is one) regarding path generation in node_fs is having an effect.

Therefore, this time I confined the implementation within node_modules instead of applying it to the whole system.

The following is the error log when applied to the entire system.
https://github.com/yamachu/node/actions/runs/12941644792/job/36098035649?pr=3#step:7:5033

code diff
https://github.com/yamachu/node/pull/3/commits/0041793aa35a6cf0c28d2334600f7e625439d77a

Reply:
Ah, I see.
Do you mean to make these changes?

util-inl.h
```cpp
#ifdef _WIN32
inline std::wstring ConvertToWideString(const std::string& str, UINT cp) {
  int size_needed = MultiByteToWideChar(
      cp, 0, &str[0], static_cast<int>(str.size()), nullptr, 0);
  std::wstring wstrTo(size_needed, 0);
  MultiByteToWideChar(
      cp, 0, &str[0], static_cast<int>(str.size()), &wstrTo[0], size_needed);
  return wstrTo;
}
#endif  // _WIN32
```

node_file.cc
```cpp
#define BufferValueToPath(str)                                                 \
  std::filesystem::path(ConvertToWideString(str.ToString(), CP_UTF8))
```

I thought that this implementation would certainly have less impact and could be implemented with less code duplication.

Reply:
Yes, exactly.

Reply:
apply 👍  : https://github.com/nodejs/node/pull/56696/commits/4ddd7aeb532653367721f779e85be03746b955f7